### PR TITLE
HomeWork_2 task_2 done

### PR DESCRIPTION
**Pull request by ivanzuev78**
Homework 2 task 2 done. 
As you advised, i had made test in the file with task.

### CODE REVIEW

File: HomeWork_2/hw2.py [hunk 1]
```diff
@@ -17,8 +17,45 @@
...
 
 """
+import unittest
 from typing import List, Tuple
 
 
 def major_and_minor_elem(inp: List) -> Tuple[int, int]:
-    ...
+
+    counter = {}
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw2.py:26]: Check the collections module there are at least two better "dicts" for that case.

File: HomeWork_2/hw2.py [hunk 2]
```diff
@@ -17,8 +17,45 @@
...
+
+    return major_elem, minor_elem
+
+
+class MyTestCase(unittest.TestCase):
+    def test_major_and_minor_elem(self):
+
+        input_1 = [2, 2, 1, 1, 1, 2, 2]
+        input_2 = [3, 2, 3]
+        input_3 = [i for i in range(20)] + [i for i in range(19)] + [10]
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw2.py:53]: Very complex, check, you can achieve the same test coverage with smalle input with out any generation.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_2/hw2.py [hunk 3]
```diff
@@ -17,8 +17,45 @@
...
+
+        self.assertEqual(major_and_minor_elem(input_1), (2, 1))
+        self.assertEqual(major_and_minor_elem(input_2), (3, 2))
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw2.py:56]: Joining multiple test to one considered a bad practice.  If first test failed, you dont know about others. 

Please move them to separate tests.

File: HomeWork_2/hw2.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_2/hw2.py]: Yes. Counter from collections is much more better. Thanks)

File: HomeWork_2/hw2.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork_2/hw2.py]: I suppose, its enought first two inputs to cover function.

File: HomeWork_2/hw2.py (hunk 3 above)
Comment (ivanzuev78) [reply HomeWork_2/hw2.py]: Done.

File: HomeWork_2/hw2.py [hunk 4]
```diff
@@ -17,8 +17,44 @@
...
+    # major_elem = None
+    minor_elem = None
+
+    for elem in inp:
+        counter[elem] += 1
+
+    major_elem = counter.most_common(1)[0][0]
+    minor_elem = counter.most_common()[-1][0]
+
+    # for elem in counter:
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw2.py:37]: What is this comment about?

File: HomeWork_2/hw2.py [hunk 5]
```diff
@@ -17,8 +17,44 @@
...
+from collections import Counter
 from typing import List, Tuple
 
 
 def major_and_minor_elem(inp: List) -> Tuple[int, int]:
-    ...
+
+    counter = Counter()
+
+    # major_elem = None
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw2.py:28]: What is this comment about?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_2/hw2.py (hunk 4 above)
Comment (ivanzuev78) [reply HomeWork_2/hw2.py:37]: Sorry, I had commented previos code and forgotten to delete it.

File: HomeWork_2/hw2.py (hunk 5 above)
Comment (ivanzuev78) [reply HomeWork_2/hw2.py]: I commented previous code and had forgotten to delete it

Comment (Cjkjvfnby) [reply HomeWork_2/hw2.py]: That kind of error could be catched by automated checks. For some reason checks were not run on this PR.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**